python/sglang/srt/managers/scheduler_output_processor_mixin.py

```diff
@@ -289,6 +294,11 @@ def process_batch_result_decode(
                     self.abort_request(AbortReq(req.rid))
                 req.grammar.finished = req.finished()
 
+        for req in batch.reqs:
+            if req.finished():
+                self.tree_cache.cache_finished_req(req)
+                req.time_stats.completion_time = time.time()
```

Comment:
I think for overlap scheduling, the same finished request will be sent to `process_batch_result_decode` multiple times, which may cause the request to be added to the tree cache multiple times. (This wasn't a problem previously since this we had a `continue` statement in the for loop if we detected that overlap was enabled and the request was finished already)